jedi/evaluate/docstrings.py

```diff
@@ -55,14 +59,49 @@ def _search_param_in_numpydocstr(docstr, param_str):
                 m = re.match('([^,]+(,[^,]+)*?)(,[ ]*optional)?$', p_type)
                 if m:
                     p_type = m.group(1)
-
-                if p_type.startswith('{'):
-                    types = set(type(x).__name__ for x in literal_eval(p_type))
-                    return list(types)
-                else:
-                    return [p_type]
+                return _expand_typestr(p_type)
         return []
 
+    def _search_return_in_numpydocstr(docstr):
+        r"""
+        Search `docstr` (in numpydoc format) for type(-s) of `param_str`.
+        """
+        doc = NumpyDocString(docstr)
+        returns = doc._parsed_data['Returns']
+        returns += doc._parsed_data['Yields']
+        found = []
+        for p_name, p_type, p_descr in returns:
+            if not p_type:
+                p_type = p_name
+                p_name = ''
+
+            m = re.match('([^,]+(,[^,]+)*?)$', p_type)
```

Comment:
Nothing I think. I removed it and it still works.

It seems to be copied from the exiting numpydocs params function:
`m = re.match('([^,]+(,[^,]+)*?)(,[ ]*optional)?$', p_type)`
which seems like a complicated way of removing the `, optional` from the parameter type string.